core/src/main/scala/com/rallyhealth/vapors/factfilter/evaluator/EvalExprAsOutputAndParam.scala

```diff
@@ -0,0 +1,105 @@
+package com.rallyhealth.vapors.factfilter.evaluator
+
+import cats.kernel.Monoid
+import cats.{Eval, FlatMap, Foldable, Functor}
+import com.rallyhealth.vapors.core.algebra.Expr
+import com.rallyhealth.vapors.core.logic.{Conjunction, Disjunction, Negation}
+import com.rallyhealth.vapors.core.math.{Addition, Negative, Subtraction}
+import com.rallyhealth.vapors.factfilter.data.{ExtractBoolean, FactSet}
+import com.rallyhealth.vapors.factfilter.evaluator.InterpretExprAsFunction.{Input, Output}
+import shapeless.HList
+
+// TODO: Make this a proxy to a function?
+class EvalExprAsOutputAndParam[F[_] : Foldable, V, P]
+  extends Expr.Visitor[F, V, P, Lambda[r => Input[F, V] => (Output[r], List[Eval[P]])]] {
```

Comment:
It sucks to have to write this class. I wonder if it is possible to write a simpler visitor that proxies to a function (or set of functions).

Reply:
BLKR: This definitely needs documentation.